lnwallet/channel.go

```diff
@@ -2604,6 +2604,18 @@ func (lc *LightningChannel) evaluateHTLCView(view *htlcView, ourBalance,
 		feePerKw: view.feePerKw,
 	}
 
+	// The fee rate of our view is always the last UpdateFee message from
+	// the channel's OpeningParty.
+	openerUpdates := view.updates.GetForParty(lc.channelState.Initiator())
+	feeUpdates := fn.Filter(func(u *paymentDescriptor) bool {
+		return u.EntryType == FeeUpdate
+	}, openerUpdates)
+	if len(feeUpdates) > 0 {
+		newView.feePerKw = chainfee.SatPerKWeight(
```

Comment:
where is the `h == 0` check?

Reply:
So the rationale for excluding it is as follows: the fee rate of an htlcView is *always the most recent fee update we have received...* irrespective of its add/remove heights.